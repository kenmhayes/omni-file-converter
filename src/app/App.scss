@import "../assets/scss/bootstrapcustom.scss";