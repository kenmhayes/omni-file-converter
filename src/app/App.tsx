import React from 'react';
import './App.scss';
import Header from './Header';
import { Outlet, } from 'react-router-dom';

/**
 * Element containing the overall layout of the web site
 * @returns 
 *   a react element
 */
function App() {
  

  return (
    <div className="App">
      <div className='container'>
        <Header />
      </div>
      <div className="container">
        <Outlet />
      </div>
    </div>
  );
}

export default App;
