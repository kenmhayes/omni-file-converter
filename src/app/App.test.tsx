import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

it('renders a navbar and content', () => {
  // A router is necessary for Links to work
  const component = create(
    <MemoryRouter>
      <App/ >
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
