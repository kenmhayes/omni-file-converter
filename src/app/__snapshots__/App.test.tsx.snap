// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a navbar and content 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <nav
      className="navbar navbar-expand-lg bg-body-tertiary"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Omni
        </a>
        <div
          className="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            className="navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Convert
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/myfiles"
                onClick={[Function]}
              >
                My Files
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="container"
  />
</div>
`;
